### DISCUSSION TITLE
Large workspace installation slow on Windows

### DISCUSSION THREAD

**Discussion by Methuselah96** [pnpm/pnpm#8489] [General]
I've been looking into switching from Yarn PnP to pnpm for a large workspace I have at work. I assumed that the installation time of Yarn PnP and pnpm would be fairly close, since that's what the [pnpm benchmarks](https://pnpm.io/benchmarks#lots-of-files) seem to suggest. However, when benchmarking the installation times on our work project, specifically on Windows, pnpm was noticeably slower than Yarn PnP.

I forked pnpm's benchmark tool, and added [a new fixture](https://github.com/Methuselah96/pnpm.io/tree/main/benchmarks/fixtures/large-workspace) that matches the workspace layout our large workspace. Here are the results I'm currently getting:

### Windows

| action  | cache | lockfile | node_modules| npm | pnpm | Yarn | Yarn PnP |
| ---     | ---   | ---      | ---         | --- | ---  | ---  | ---      |
| install |       |          |             | 1m 31.4s | 41.3s | 38.4s | 17.3s |
| install | ✔     | ✔        | ✔           | 3.8s | 2.6s | 3.3s | n/a |
| install | ✔     | ✔        |             | 39.5s | 18.4s | 23.9s | 2.7s |
| install | ✔     |          |             | 55.8s | 29.1s | 27.7s | 5.7s |
| install |       | ✔        |             | 45.5s | 31.9s | 33.6s | 13.1s |
| install | ✔     |          | ✔           | 4.3s | 7.6s | 6.5s | n/a |
| install |       | ✔        | ✔           | 4.6s | 2.6s | 13.2s | n/a |
| install |       |          | ✔           | 4.3s | 26.5s | 16s | n/a |

<img src="https://raw.githubusercontent.com/Methuselah96/pnpm.io/999df17c01baa4b643a8fe721e3ce3353a5406b8/static/img/benchmarks/large-workspace-Windows.svg" />

### Linux

| action  | cache | lockfile | node_modules| npm | pnpm | Yarn | Yarn PnP |
| ---     | ---   | ---      | ---         | --- | ---  | ---  | ---      |
| install |       |          |             | 1m 9.6s | 20.2s | 26.6s | 14.3s |
| install | ✔     | ✔        | ✔           | 3.3s | 1.8s | 2.6s | n/a |
| install | ✔     | ✔        |             | 18.7s | 5.2s | 14.5s | 2.2s |
| install | ✔     |          |             | 30.2s | 14.3s | 18.1s | 5.1s |
| install |       | ✔        |             | 24.3s | 11.1s | 22.6s | 10.1s |
| install | ✔     |          | ✔           | 3s | 6.9s | 5.3s | n/a |
| install |       | ✔        | ✔           | 3.1s | 1.9s | 10.8s | n/a |
| install |       |          | ✔           | 3s | 13.3s | 14s | n/a |

<img src="https://raw.githubusercontent.com/Methuselah96/pnpm.io/0c8e962ded9432fb9114b4b88f2eb06a9e40ff7a/static/img/benchmarks/large-workspace-Linux.svg" />

I'm assuming based on the fact that Windows is much slower than Linux, that the slow installation time is due to Windows file system operations being slow.

Has anyone else experienced this? Is there anything that can be done to make the package installation faster on Windows?

**Comment by AnderssonPeter** [comment]
For us `pnpm` is faster than `npm` except on a clean install with no cached packages in the store.
npm: 120 s
pnpm: 300 s

pnpm is more than two times as slow as normal npm is this normal? (we are running on windows)
